XiEditor/EditViewController.swift

```diff
@@ -107,7 +112,7 @@ class EditViewController: NSViewController, EditViewDataSource {
         let first = Int(floor(scrollView.contentView.bounds.origin.y / textMetrics.linespace))
         let height = Int(ceil((scrollView.contentView.bounds.size.height) / textMetrics.linespace))
         let last = first + height
-        if first != firstLine || last != lastLine && document != nil {
```

Comment:
Great. If It turns out I'm wrong I will happily work on some alternative solution. 👍